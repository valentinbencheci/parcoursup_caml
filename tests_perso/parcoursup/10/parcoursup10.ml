(* ###################### - DESCRIPTION TEST - ###################### *)
(* On regarde comment l'algo reagit si on donne une voeu de rang None 
et apres de rang negatif(-1) *)
(* ################################################################## *)

open Test_parcoursup_utils

let session = nouvelle_session ()

let () =
  ajoute_candidat session ~nom_candidat:"Adam";

  ajoute_formation session ~nom_formation:"Université Côte d'Azur" ~capacite:1;

  ajoute_voeu session
    ~rang_repondeur:(None)
    ~nom_candidat:"Adam"
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session
    ~rang_repondeur:(Some (-1))
    ~nom_candidat:"Adam"
    ~nom_formation:"Université Côte d'Azur";

  ajoute_commission session
    ~nom_formation:"Université Côte d'Azur"
    ["Adam"];

  reunit_commissions session;

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";

  renonce session
    ~nom_candidat:"Adam"
    ~nom_formation:"Université Côte d'Azur";

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";