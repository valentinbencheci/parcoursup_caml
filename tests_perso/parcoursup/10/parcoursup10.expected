nouvelle session
ajoute candidat Adam
ajoute formation Université Côte d'Azur (1 places)
ajoute voeu de Adam pour Université Côte d'Azur
ajoute voeu de Adam pour Université Côte d'Azur (rang=-1)
commission de classement de Université Côte d'Azur : [Adam]
réunit commissions
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université Côte d'Azur}
Adam renonce à Université Côte d'Azur
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {}
